x-pack/elastic-agent/pkg/agent/application/config.go

```diff
@@ -26,3 +33,50 @@ func createFleetConfigFromEnroll(accessAPIKey string, kbn *kibana.Config) (*conf
 	}
 	return cfg, nil
 }
+
+// LoadConfigFromFile loads the Agent configuration from a file.
+//
+// This must be used to load the Agent configuration, so that variables defined in the inputs are not
+// parsed by go-ucfg. Variables from the inputs should be parsed by the transpiler.
+func LoadConfigFromFile(path string) (*config.Config, error) {
```

Comment:
Not official supported but it basically means someone can use `${ELASTIC_USERNAME}` today in the output config because it is ucfg but in case we also support static providers like keystore in the output in the future, this might break as it needs a prefix?

Reply:
Yes this allows the same go-ucfg that we have always allowed anywhere in the config except in the `inputs`. But if we change that in the future then a prefix would be required to get it from the keystore provider.

Reply:
Lets keep it in mind if we ever there but ok for now.